nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AbstractRecordProcessor.java

```diff
@@ -174,6 +174,7 @@ public void process(final InputStream in, final OutputStream out) throws IOExcep
             });
         } catch (final Exception e) {
             getLogger().error("Failed to process {}; will route to failure", new Object[] {flowFile, e});
+            session.putAttribute(flowFile,"record.error.message",e.getCause().getLocalizedMessage());
```

Comment:
Is it possible that `getCause()` will return null here? In that case we could probably return `e.getLocalizedMessage()`, and if either message is null, we should probably return the class name of the exception/cause.

Reply:
We're wrapping both exceptions directly above here so I think they're should always be a cause but it's possible the message might be null.